### PR TITLE
Add rtcl::rupdate

### PR DESCRIPTION
**Pull request by niheaven**
- R tools for the command line
- https://github.com/rdatsci/rtcl
- Like Jetpack, but more powerful.

**`rupdate`'s a sub command from `rtcl` package, and I don't know how to properly name it in topgrade**

## Standards checklist:

- [x] The PR title is descriptive.
- [x] The code compiles
- [x] The code passes rustfmt
- [x] The code passes clippy
- [x] The code passes tests
- [x] *Optional:* I have tested the code myself
    - [x] I also tested that Topgrade skips the step where needed

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -280,6 +280,7 @@ fn run() -> Result<()> {
     runner.execute(Step::Krew, "krew", || generic::run_krew_upgrade(run_type))?;
     runner.execute(Step::Gem, "gem", || generic::run_gem(&base_dirs, run_type))?;
     runner.execute(Step::Sheldon, "sheldon", || generic::run_sheldon(&ctx))?;
+    runner.execute(Step::Rtcl, "rtcl", || generic::run_rtcl(&ctx))?;
```
Comment (r-darwish) [inline src/main.rs:283]: Could you please tell me which tool is this about? I could not find it.

**Review by r-darwish — CHANGES_REQUESTED** [review]

File: src/main.rs (hunk 1 above)
Comment (niheaven) [reply src/main.rs:283]: `rtcl` is just **R tools for the command line**, and `rupdate` is one of its command, located in packages' `bin` or `win` subdir.

> `rupdate` to update all CRAN packages on your system, install missing CRAN packages listed in your collection `~/.rtcl/packages` and update all packages with a git source.

`jetpack` only update packages that installed from CRAN, `rtcl::rupdate` will update packages installed from CRAN and git resources (like GitHub, etc.).

The package is not on CRAN yet, and need to be installed from https://github.com/rdatsci/rtcl.

File: src/steps/generic.rs [hunk 2]
```diff
@@ -121,6 +121,14 @@ pub fn run_jetpack(run_type: RunType) -> Result<()> {
     run_type.execute(&jetpack).args(&["global", "update"]).check_run()
 }
 
+pub fn run_rtcl(ctx: &ExecutionContext) -> Result<()> {
+    let rupdate = utils::require("rupdate")?;
+
+    print_separator("rtcl::rupdate");
```
Comment (r-darwish) [inline src/steps/generic.rs:127]: Please put a friendlier name

Comment (niheaven) [reply src/steps/generic.rs:127]: Yes, that what's I'm not sure... Do you think in this case `rupdate` or just `rtcl` be a good name?

Comment (r-darwish) [reply src/steps/generic.rs:127]: I think rtctl is a better choice

Comment (niheaven) [reply src/steps/generic.rs]: Using `rtcl` as separator.

**Comment by r-darwish** [comment]
Thanks!

**Pull request merged**